File: Java General/CANdle MultiAnimation/src/main/java/frc/robot/commands/CANdlePrintCommands.java
package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.InstantCommand;
import frc.robot.subsystems.CANdleSystem;

public class CANdlePrintCommands {
    static public class PrintVBat extends InstantCommand {
        public PrintVBat(CANdleSystem candleSystem) {
            super(() -> System.out.println("VBat is " + candleSystem.getVbat() + " Vs"), candleSystem);
        }
        @Override
        public boolean runsWhenDisabled() {
            return true;
        }
    }
    static public class Print5V extends InstantCommand {
        public Print5V(CANdleSystem candleSystem) {
            super(() -> System.out.println("5V is " + candleSystem.get5V() + "V"), candleSystem);
        }
        @Override
        public boolean runsWhenDisabled() {
            return true;
        }
    }
    static public class PrintCurrent extends InstantCommand {
        public PrintCurrent(CANdleSystem candleSystem) {
            super(() -> System.out.println("Current is " + candleSystem.getCurrent() + "A"), candleSystem);
        }
        @Override
        public boolean runsWhenDisabled() {
            return true;
        }
    }
    static public class PrintTemperature extends InstantCommand {
        public PrintTemperature(CANdleSystem candleSystem) {
            super(() -> System.out.println("Temperature is " + candleSystem.getTemperature() + "C"), candleSystem);
        }
        @Override
        public boolean runsWhenDisabled() {
            return true;
        }
    }
}

File: C++ General/CANdle-MultiAnimation/src/main/cpp/commands/CANdlePrintCommands.cpp
#include "commands/CANdlePrintCommands.h"
#include <iostream>

using namespace CANdlePrintCommands;

PrintVBat::PrintVBat(CANdleSystem *candleSystem) :
    frc2::InstantCommand([this, candleSystem]{ 
        std::cout << "VBat is " << candleSystem->GetVbat() << "V" << std::endl;
    }) {
    
}

Print5V::Print5V(CANdleSystem *candleSystem) :
    frc2::InstantCommand([this, candleSystem]{ 
        std::cout << "5V is " << candleSystem->Get5V() << "V" << std::endl;
    }) {
    
}

PrintCurrent::PrintCurrent(CANdleSystem *candleSystem) :
    frc2::InstantCommand([this, candleSystem]{ 
        std::cout << "Current is " << candleSystem->GetCurrent() << "A" << std::endl;
    }) {
    
}

PrintTemperature::PrintTemperature(CANdleSystem *candleSystem) :
    frc2::InstantCommand([this, candleSystem]{ 
        std::cout << "Temperature is " << candleSystem->GetTemperature() << "C" << std::endl;
    }) {
    
}


PrintModulatedOutput::PrintModulatedOutput(CANdleSystem *candleSystem) :
    frc2::InstantCommand([this, candleSystem]{ 
        std::cout << "Modulated Output is " << candleSystem->GetModulatedOutput() * 100 << "%" << std::endl;
    }) {
    
}

